src/persist/src/indexed/encoding.rs

```diff
@@ -194,20 +194,35 @@ impl BlobMeta {
             }
             ids.insert(*id);
         }
+        
+        let mut futures = HashMap::new();
         for (id, f) in self.futures.iter() {
             if !ids.contains(id) {
                 return Err(format!("futures id {:?} not present in id_mapping", id).into());
             }
+            
+            if futures.contains_key(id) {
+                return Err(format!("duplicate future: {:?}", id).into());
+            }
+
             f.validate()?;
+            futures.insert(*id, f);
```

Comment:
super nit: insert this above the valiate call to keep it next to the contains check

Reply:
done!